lnrpc/websocket_proxy.go

```diff
@@ -239,12 +272,17 @@ func (p *WebsocketProxy) upgradeToWebSocketProxy(w http.ResponseWriter,
 
 		// Whenever a pong message comes in, we extend the deadline
 		// until the next read is expected by the interval plus pong
-		// wait time.
+		// wait time. Since we can never _reach_ any of the deadlines,
+		// we also have to advance the deadline for the next expected
+		// write to happen, in case the next thing we actually write is
+		// the next ping.
 		conn.SetPongHandler(func(appData string) error {
 			nextDeadline := time.Now().Add(
 				p.pingInterval + p.pongWait,
 			)
 			_ = conn.SetReadDeadline(nextDeadline)
+			_ = conn.SetWriteDeadline(nextDeadline)
```

Comment:
Shouldn't we actually be setting the write deadline when we write out _either_ the ping or any other write message? AFAICT, here when we get a new pong message we'll reset both deadlines, which seems wrong, as we actually haven't written anything here.

Reply:
So we should have a modification of the write deadline below when we do `responseForwarder.Scan() {`.

Reply:
I need to re-absorb the greater control here methinks

Reply:
We could do that, yes. But I think for understanding the logic a bit better it's sufficient to only use/set deadlines if we're actually using the ping/pong mechanism.

Reply:
We'd need to wrap that `SetDeadline()` call in the `responseForwarder.Scan()` loop with an `if p.pingPongEnabled() {`. Otherwise we'd set a deadline but aren't guaranteed to have something written until then. So I think it just makes it harder to understand and doesn't _add_ anything.